pylint/checkers/typecheck.py

```diff
@@ -381,6 +381,12 @@ def _missing_member_hint(
         "Used when a slice step is 0 and the object doesn't implement "
         "a custom __getitem__ method.",
     ),
+    "E1145": (
```

Comment:
```suggestion
    "E1145": (
```
Should not be an error imo, but I also think it should not be a message either. Or at least some prior design is in order. Isn't the goal to auto-populate the function-mutator option based on comment by the lib authors directly in their code ?  I like this the end goal, but this would be something new in pylint, with the expected associated maintenance cost. Maybe a hack like this is using the existing message control framework is good, maybe it would be simpler to suggest to add a value to ``function-mutator`` manually in the message ? This is not something that is done often, even less so if we do configuration template, this way it's done once per library and become very reasonable.

Reply:
> Should not be an error imo, but I also think it should not be a message either.

Agreed. I just needed a simple way to hook into the inline comment system. It feels like this comment is meant to "mark" decorator definitions as being signature-mutators. Thus, a more ideal comment might read:

```python
def my_decorator(func): # pylint: mark=signature-mutator
    ...
```

I'm not really sure how to do this though.

> Maybe it would be simpler to suggest to add a value to function-mutator manually in the message?

If I understand this correctly you're suggesting that, if PyLint discovers invalid arguments, and the function in question has a decorator, that PyLint report the error, but also recommend adding the decorator to the `signature-mutators` list manually. If so, I think this could be problematic. Signature mutating decorators are a fairly advanced concept and users might add decorators to that list without fully understanding what they're doing. I foresee bug reports to PyLint from people who have done this and are getting false negatives for invalid function calls. Further, as a lib author, even with this addition to the error message, I'm still probably going to get bug reports from people complaining about this PyLint false positive. Ultimately, as a lib author myself, I didn't want to deal with that and [chose to just not use the decorator syntax](https://github.com/idom-team/idom/blob/main/src/idom/html.py#L400-L446).

Reply:
I don't have any useful suggestions on implementation (sorry), but just to add a bit of weight to the cause... I'm coming from a completely unrelated project from @rmorshea and was independently looking for a solution to the exact same problem. I agree with everything he's said here. I understand there's no other patterns like this in pylint at the moment, but it would be much, much more convenient if this `disable` could be somehow specified at point of function definition rather than at point of call. I'm developing a package that defines decorated functions that will unfortunately raise `unexpected-keyword-arg` when called. At the moment there's a burden on every user that uses pylint and calls my function to update their config; it would be much more maintainable if the burden fell on the package author instead.

Reply:
Perhaps rather than creating a new pattern, should pylint consider disabling this error on all decorated functions?

In theory, this change should only be a temporary until a method is determined on how to properly parse and analyze these functions.

Reply:
We have ``function-mutator`` to handle this. Some functions are just too dynamic and cannot reasonably be parsed but that's not all functions and pylint would be crippled if we just refuse to analyses all decorators. Maybe we could suggest to populate ``function_mutator`` only on decorator we're not sure about. Maybe we can add more default values in function-mutator, for well known libs.

Reply:
Brutal opinion here: If I need to start populating a list of decorated functions within pylintrc, I'd likely remove pylint from my workflow instead.